shell/browser/api/electron_api_base_window.h

```diff
@@ -274,7 +274,7 @@ class BaseWindow : public gin_helper::TrackableObject<BaseWindow>,
   // Reference to JS wrapper to prevent garbage collection.
   v8::Global<v8::Value> self_ref_;
 
-  base::WeakPtrFactory<BaseWindow> weak_factory_;
+  base::WeakPtrFactory<BaseWindow> weak_factory_{this};
```

Comment:
NB for reviewers: since the parent class [is initialized before](https://stackoverflow.com/a/40633475) these members during construction, this should be safe

Reply:
also we've been using this construct in some classes already